src/test/java/com/walmartlabs/x12/util/split/X12TransactionSplitterTest.java

```diff
@@ -0,0 +1,426 @@
+        assertNotNull(ediDocuments);
+        assertEquals(0, ediDocuments.size());
+    }
+    
+    @Test
+    public void test_split_sourceData_missing_SE() throws IOException {
+        String sourceData = new StringBuilder()
+            .append("ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ*123456789012345*101127*1719*U*00400*000000049*0*P*>")
+            .append("\r\n")
+            .append("GS*SH*4405197800*999999999*20111206*1045*00*X*004060")
+            .append("\r\n")
+            .append("ST*AAA*0001")
+            .append("\r\n")
+            .append("TEST*1")
+            .append("\r\n")
+            // missing SE
+            .append("GE*1*00")
+            .append("\r\n")
+            .append("IEA*2*000000049")
+            .toString();
+        
+        try {
+            splitter.split(sourceData);
+            fail("expected X12ParserException");
+        } catch (X12ParserException e) {
+            assertEquals("expected SE segment but got IEA", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void test_split_sourceData_missing_GE() throws IOException {
+        String sourceData = new StringBuilder()
+            .append("ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ*123456789012345*101127*1719*U*00400*000000049*0*P*>")
+            .append("\r\n")
+            .append("GS*SH*4405197800*999999999*20111206*1045*00*X*004060")
+            .append("\r\n")
+            .append("ST*AAA*0001")
+            .append("\r\n")
+            .append("TEST*1")
+            .append("\r\n")
+            .append("SE*1*0001")
+            .append("\r\n")
+            //missing GE
+            .append("IEA*2*000000049")
+            .toString();
+        
+        try {
+            splitter.split(sourceData);
+            fail("expected X12ParserException");
+        } catch (X12ParserException e) {
+            assertEquals("expected ST segment but got IEA", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void test_split_sourceData_missing_ST() throws IOException {
+        String sourceData = new StringBuilder()
+            .append("ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ*123456789012345*101127*1719*U*00400*000000049*0*P*>")
+            .append("\r\n")
+            .append("GS*SH*4405197800*999999999*20111206*1045*00*X*004060")
+            .append("\r\n")
+            // missing ST
+            .append("TEST*1")
+            .append("\r\n")
+            .append("SE*1*0001")
+            .append("GE*1*00")
+            .append("\r\n")
+            .append("IEA*2*000000049")
+            .toString();
+        
+        try {
+            splitter.split(sourceData);
+            fail("expected X12ParserException");
+        } catch (X12ParserException e) {
+            assertEquals("expected ST segment but got TEST", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void test_split_sourceData_two_documents_missing_SE_blend() throws IOException {
```

Comment:
Shouldn't this test case fail? 🤔  Is it passing because not all code from #86 present here?

Reply:
the splitter is relatively "dumb" so the SE on line 154 is taken as the end of the transaction started on line 145. 
this means the two transactions are treated as one larger one. 

right now, the principle is pass the split transaction out to another process to evaluate. 

that said ... 🤔 
maybe can throw error if get another ST before an SE